### PR TITLE
Implement AttestationAndProof DS (v0.9.0)

### PR DESCRIPTION
**Pull request by cemozerr**
## PR Description
This PR adds the AggregateAndProof data structure and implements its ser/deser tooling.

### CODE REVIEW

File: ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/operations/AggregateAndProof.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  private Attestation aggregate;
+
+  public AggregateAndProof(
+      UnsignedLong index, BLSSignature selection_proof, Attestation aggregate) {
+    this.index = index;
+    this.selection_proof = selection_proof;
+    this.aggregate = aggregate;
+  }
+
+  public AggregateAndProof() {
```
Comment (mbaxter) [inline ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/operations/AggregateAndProof.java:44]: Do we need this empty constructor + setters?  Or can we get away with an immutable object?

Comment (cemozerr) [reply ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/operations/AggregateAndProof.java:44]: I think we can. Let me make that change real quickly.

File: validator/client/src/main/java/tech/pegasys/artemis/validator/client/ValidatorAssignment.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package tech.pegasys.artemis.validator.client;
+
+public class ValidatorAssignment {
```
Comment (ajsutton) [inline validator/client/src/main/java/tech/pegasys/artemis/validator/client/ValidatorAssignment.java]: Maybe just omit this class for now?

**Review by ajsutton — APPROVED** [review]
LGTM.  Just need to run spotless to make the build happy I suspect.

File: validator/client/src/main/java/tech/pegasys/artemis/validator/client/ValidatorAssignment.java (hunk 2 above)
Comment (cemozerr) [reply validator/client/src/main/java/tech/pegasys/artemis/validator/client/ValidatorAssignment.java]: oh I must have missed that. done.

**Pull request merged**